json_yaml_files/prepare_files.py

```diff
@@ -0,0 +1,184 @@
+            return self.scan_type[Y][Z.lower()]
+        if '_' in Z:
+            prefix = Z.split('_')[0]
+            if prefix.lower() in self.scan_type.get(Y, {}):
+                return self.scan_type[Y][prefix.lower()]
+        return self.scan_type.get(Y, {}).get(Y, '')
+
+
+    def prepare_yaml_contents(self, file_name):
+        _, _, Y, Z = self.extract_file_name_components(file_name)
+
+        yaml_contents = {
+            "scan_type": self.fetch_scan_type(Y, Z),
+            "scan_object": 'Live',
+            "image_file_format": 'NIFTI',
+            "image_modality": self.image_modality[Y],
+            "transformation_performed": 'No'
+        }
+        output_file = self.yaml_dir / file_name
+        self._write_to_file(output_file, yaml_contents, file_format='yaml')
+
+
+    def fetch_Z_value(self, files):
+        Z = set()
+        ignore_list = ['desc', 'subject', 'session', 'extension', 'suffix', 'datatype']
+        
+        for file in files:
+            entities = self.layout.parse_file_entities(file)
+            filtered_entities = {k: v for k, v in entities.items() if k not in ignore_list}
+            
+            for key, value in filtered_entities.items():
+                z = f"{key}-{value}" if key != 'space' else value
+                if entities.get('suffix') and entities.get('suffix') != entities.get('datatype'):
+                    z = f"{z}_{entities['suffix']}" if z else entities['suffix']
+                if z:
+                    Z.add(z) 
+            if not Z and entities['suffix'] != entities['datatype']:
+                Z.add(entities['suffix'])
+                    
+        return list(Z)
+
+    def prepare_file_names(self, X, Y, Z):
+        file_base = f"{self.A}_{X}.{Y}."
+        return {
+            'json': [f"{file_base}{z}.json" for z in Z],
+            'yaml': [f"{file_base}{z}.yaml" for z in Z]
+        }
+    
+    
+    def _replace_placeholders(self, path, subject, session):
+        if subject:
+            path = path.replace(subject, '{SUBJECT}')
+        if session:
+            path = path.replace(session, '{SESSION}')
+        return re.sub(r'-\d', '-{#}', path)
+    
+    def _write_to_file(self, output_file, contents, file_format='json'):
+        with open(output_file, 'w') as f:
+            if file_format == 'yaml':
+                yaml.dump(contents, f)
+            else:
+                json.dump(contents, f, indent=2)
+                
+                
+    def run(self): 
+        scopes = {
+            'raw': 'inputs',
+            'derivatives': 'derivatives',
+            'sourcedata': 'sourcedata'
+        }
+        X = [name for scope, name in scopes.items() if self.layout.get(scope=scope)]
+        Y_types = self.layout.get_datatypes()
+        
+        for x in X:
+            for y in Y_types:
+                files = self.layout.get(
+                    scope=x if x != 'inputs' else 'raw', datatype=y, return_type='file')
```

Comment:
Please expand this line into multiple lines for easier reading. Specifically:

`x if x != 'inputs' else 'raw'`

Reply:
Sure I will do that.